src/averageAges.js

```diff
@@ -56,6 +81,21 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const mothers = new Set(people.map(person => person.name));
+
+  const kids = onlyWithSon
+    ? people.filter(child => child.sex === 'm'
+    && mothers.has(child.mother))
+    : people.filter(child => mothers.has(child.mother));
+
+  const difference = kids.map(
+    child => child.born - people.find(
+      mother => mother.name === child.mother).born
+  );
```

Comment:
If you map over kids array, the iterator variable should also be `kid` (refering to the array name), or name the array children
```suggestion
  const ageDifference = kids.map(
    kid => kid.born - people.find(
      mother => mother.name === kid.mother).born
  );
```